src/mSupplyMobileApp.js

```diff
@@ -308,8 +321,66 @@ class MSupplyMobileAppContainer extends React.Component {
                 : this.renderSyncState
             }
           />
-          <MainStackNavigator.Navigator>
-            <MainStackNavigator.Screen name="root" component={MenuPage} />
+          <MainStackNavigator.Navigator
+            initialRouteName={ROUTES.ROOT}
+            screenOptions={{ headerShown: false }}
+          >
+            <MainStackNavigator.Screen name={ROUTES.ROOT} component={MenuPage} />
+
+            <MainStackNavigator.Screen
+              name={ROUTES.CUSTOMER_REQUISITION}
+              component={CustomerRequisitionPage}
+            />
+            <MainStackNavigator.Screen
+              name={ROUTES.CUSTOMER_REQUISITIONS}
+              component={CustomerRequisitionsPage}
+            />
+
+            <MainStackNavigator.Screen
+              name={ROUTES.SUPPLIER_REQUISITION}
+              component={SupplierRequisitionPage}
+            />
+            <MainStackNavigator.Screen
+              name={ROUTES.SUPPLIER_REQUISITIONS}
+              component={SupplierRequisitionsPage}
+            />
+
+            <MainStackNavigator.Screen
+              name={ROUTES.SUPPLIER_INVOICE}
+              component={SupplierInvoicePage}
+            />
+            <MainStackNavigator.Screen
+              name={ROUTES.SUPPLIER_INVOICES}
+              component={SupplierInvoicesPage}
+            />
+
+            <MainStackNavigator.Screen
+              name={ROUTES.CUSTOMER_INVOICE}
+              component={CustomerInvoicePage}
+            />
+            <MainStackNavigator.Screen
+              name={ROUTES.CUSTOMER_INVOICES}
+              component={CustomerInvoicesPage}
+            />
+
+            <MainStackNavigator.Screen name={ROUTES.STOCK} component={StockPage} />
+
+            <MainStackNavigator.Screen name={ROUTES.STOCKTAKES} component={StocktakesPage} />
+            <MainStackNavigator.Screen
+              name={ROUTES.STOCKTAKE_MANAGER}
+              component={StocktakeManagePage}
+            />
+            <MainStackNavigator.Screen
+              name={ROUTES.STOCKTAKE_EDITOR}
+              component={StocktakeEditPage}
+            />
+
+            <MainStackNavigator.Screen name={ROUTES.DISPENSARY} component={DispensingPage} />
+            <MainStackNavigator.Screen name={ROUTES.PRESCRIPTION} component={PrescriptionPage} />
+
+            <MainStackNavigator.Screen name={ROUTES.REALM_EXPLORER} component={RealmExplorer} />
+            <MainStackNavigator.Screen name={ROUTES.SETTINGS} component={SettingsPage} />
+            <MainStackNavigator.Screen name={ROUTES.DASHBOARD} component={DashboardPage} />
```

Comment:
Just due to the number of components here, might be nice to programmatically generate these, e.g.:

```
ROUTES.map(route => <MainStackNavigator.Screen name={route} component={COMPONENTS[route]} />
```

Reply:
See PR comment regarding this ! waiting a bit to see if there is something nicer that can be done, potentially